builder/amazon/common/step_run_spot_instance.go

```diff
@@ -227,6 +228,33 @@ func (s *StepRunSpotInstance) Run(ctx context.Context, state multistep.StateBag)
 	}
 	instanceId = *spotResp.SpotInstanceRequests[0].InstanceId
 
+	// Tag spot instance request
+	spotTags, err := s.SpotTags.EC2Tags(s.Ctx, *ec2conn.Config.Region, state)
+	if err != nil {
+		err := fmt.Errorf("Error tagging spot request: %s", err)
+		state.Put("error", err)
+		ui.Error(err.Error())
+		return multistep.ActionHalt
+	}
+	spotTags.Report(ui)
+
+	if len(spotTags) > 0 && s.SpotTags.IsSet() {
+		// Retry creating tags for about 2.5 minutes
```

Comment:
How was the time window for this determined?

Reply:
It's a common timeout throughout the codebase.

Reply:
I'm fine to set it to any other value, but 2.5 minutes seems like a nice compromise that'll allow some time for the API call to go through - even in the worst case - while not waiting forever.

Reply:
That seems OK then.